File: CompetitiveProgramming/DMOJ/BlueBook/BlueBook - Min/2021-08-11/Min.cpp
/*
------------------------------------------------------------------------
---------------------------- BlueBook - Min ----------------------------
Find the min of a list of real numbers.
1 <= Size of list <= 10 000
-1000 < Size of numbers < 1000
Output answer to two decimal places.

Sample Input
2
10.1
12.3

Sample Output
10.10

------------------------------------------------------------------------
---------------------------- SOLUTION LOGIC ----------------------------
Use built-in min function or loop through list.

--------------------------------------------------------------------------
---------------------------- PROGRAM COMMENTS ----------------------------
Finds the min of a list of real numbers.
@author Vincent Zhang
@since 11 August 2021
*/

#include <iostream>
#include <iomanip>

/**
 * Solves the problem.
 * @return 0 Succesful program termination
 */
int main()
{
    // Disable synchronization between C and C++ standard streams allowing C++ streams independent buffers.
    std::ios_base::sync_with_stdio(false);
    // Untie cin stream from cout stream to disable automatic flushing after each IO operation on the other stream.
    std::cin.tie(nullptr);

    int n;
    std::cin >> n;

    float min = 100000;

    for(int i=0; i<n; i++)
    {
        float f;
        std::cin >> f;
        if(f < min)
            min = f;
    }
    std::cout << std::fixed << std::setprecision(2) << min << '\n';

    return 0;
}

File: CompetitiveProgramming/DMOJ/BlueBook/BlueBook - Min/2021-08-11/Min.java
/*
------------------------------------------------------------------------
---------------------------- BlueBook - Min ----------------------------
Find the min of a list of real numbers.
1 <= Size of list <= 10 000
-1000 < Size of numbers < 1000
Output answer to two decimal places.

Sample Input
2
10.1
12.3

Sample Output
10.10

------------------------------------------------------------------------
---------------------------- SOLUTION LOGIC ----------------------------
Use built-in min function or loop through list.
*/

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

/**
 * Finds the min of a list of real numbers.
 * @author Vincent Zhang
 * @since 11 August 2021
 */
public class Min
{
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    /**
     * Attempt to return next input line as integer.
     * @return next integer input line
     */
    static int nextInt()
    {
        try
        {
            return Integer.parseInt(br.readLine());
        }
        catch(IOException | NumberFormatException e)
        {
            e.printStackTrace();
        }
        return -1;
    }

    /**
     * Attempt to return next input line as float.
     * @return float input line
     */
    static float nextFloat()
    {
        try
        {
            return Float.parseFloat(br.readLine());

        }
        catch(IOException | NumberFormatException e)
        {
            e.printStackTrace();
        }
        return -1;
    }

    /**
     * Close BufferedReader.
     * @return void
     */
    static void closeBr()
    {
        try
        {
            br.close();
        }
        catch(IOException ioe)
        {
            ioe.printStackTrace();
        }
    }

    /**
     * Solves the problem.
     * @param args Command-line arguments
     * @return void
     */
    public static void main(String[] args)
    {
        int n = nextInt();
        float min = 100000;
        for(int i=0; i<n; i++)
        {
            float f = nextFloat();
            if(min > f)
            {
                min = f;
            }
        }
        closeBr();
        System.out.printf("%.2f%n", min);

    }
}